modules/core/src/main/scala/ast.scala

```diff
@@ -10,7 +10,12 @@ object Ast {
   sealed trait Definition
 
   sealed trait ExecutableDefinition extends Definition
-  sealed trait TypeSystemDefinition extends Definition
+  sealed trait TypeSystemDefinitionOrExtension extends Definition
+
+  sealed trait TypeSystemDefinition extends TypeSystemDefinitionOrExtension
+  sealed trait TypeSystemExtension extends TypeSystemDefinitionOrExtension
+
+  sealed trait TypeExtension extends TypeSystemExtension
```

Comment:
I was copying the hierarchy in the graphql spec, I guess extensions don't define a type by themselves so are arguable not `TypeDefinition`s, and I think when it comes to `Schema` it will be handy to be able to treat extensions as a different kind of thing.  That said happy to change it up if desired :)

Reply:
No, you're absolutely right.